### ISSUE TITLE
Accessing `class_servers_api` from editor triggers assertion failure after hot reloading

### ISSUE THREAD

**Issue by kang-sw** [godot-rust/gdext#629]
# Context

- Environment
  - Windows 11
  - `gdext` master branch Feb 26. 2024
  - Godot 4.2.1
- From any `tool` class which accesses to `class_servers_api` indirectly via `DisplayServer` or `RenderingServer`
- After hot reloading, any access to `class_servers_api` triggers following assertion failure(always).

```rust
// in "godot-ffi/src/binding/mod.rs:245"
pub unsafe fn class_scene_api() -> &'static ClassSceneMethodTable {
    // SAFETY: `get_binding` has the same preconditions as this function.
    let binding = unsafe { get_binding() };

    debug_assert!(
        binding.class_scene_method_table.is_initialized(), // <<< THIS ONE !!! <<<
        "cannot fetch classes; init level 'Scene' not yet loaded"
    );

    // SAFETY: `initialize_class_scene_method_table` has been called.
    unsafe { binding.class_scene_method_table.get_unchecked() }
}

```

This is a MRE for reproducing assertion failure on hot reloading. To reproduce, 
1. Build gdextension (`cargo build` at project root)
2. Open Editor(`godot/`). (Attaching debugger is recommended)
3. Open `outputs` tab to check if log message is correctly output
4. Rebuild gdextension, repeating step 1. (You may need to touch `lib.rs` file to invalidate build)
5. Back to editor to trigger hot reloading. It'll reproduce assertion failure.

[mre.zip](https://github.com/godot-rust/gdext/files/14401430/mre.zip)

---

By following above steps:

1. It'll report window position every 5 seconds by accessing display server

Code:
```rust
#[godot_api]
impl IEditorPlugin for TestEditorPlugin {
    fn process(&mut self, dt: f64) {
        self.time += dt;

        if self.time < self.next_trigger {
            return;
        }

        self.next_trigger = self.time + 5.0;

        let gd_ds = DisplayServer::singleton();

        // By moving window, you can see log output changes.
        godot_print!(
            "{:.5} Window Position: {:?}",
            self.time,
            gd_ds.window_get_position() // TRIGGERS ASSERTION
        );
    } 

    // .. skipped
}

```

Output:
![image](https://github.com/godot-rust/gdext/assets/32210953/701d9e44-1579-432e-abbd-488595a043d5)

3. Rebuild and hot reload. Right after hot reloading, the assertion failure triggers.

Call stack from _process ~ to class_servers_api assertion
```
core::panicking::panic_fmt() Line 72 (core\src\panicking.rs:72)
[Inline Frame] godot_ffi::binding::class_servers_api() Line 225 (godot-ffi\src\binding\mod.rs:225)
godot_core::gen::classes::display_server::re_export::DisplayServer::window_get_position_full(int self) Line 1238 (godot-core\src\gen\classes\display_server.rs:1238)
godot_core::gen::classes::display_server::ExWindowGetPosition::done(godot_core::gen::classes::display_server::ExWindowGetPosition) Line 2919 (godot-core\src\gen\classes\display_server.rs:2919)
godot_core::gen::classes::display_server::re_export::DisplayServer::window_get_position() Line 1245 (godot-core\src\gen\classes\display_server.rs:1245)
hot_reload_test::impl$11::process(hot_reload_test::TestEditorPlugin * self, double dt) Line 37 (gdext-hot-reload-crash\src\lib.rs:37)
```

Class Defintion:
```rust
#[derive(GodotClass)]
#[class(init, tool, editor_plugin, base=EditorPlugin)]
pub struct TestEditorPlugin {
    base: Base<EditorPlugin>,

    time: f64,
    next_trigger: f64,
}
```

At the point of assertion is after the class object is recreated; `self.time` and `self.next_trigger` are reset to 0.
![image](https://github.com/godot-rust/gdext/assets/32210953/6af1356b-df15-472e-95fa-f7cf7cb9b829)

**Comment by kang-sw** [comment]
Additionally, applying #628 still panicks at same method, but with slightly different content.

From:
```rust
    debug_assert!(
        binding.class_server_method_table.is_initialized(), // <<< THIS ONE!
        "cannot fetch classes; init level 'Servers' not yet loaded"
    );
```

To:
```rust
    debug_assert!(
        table.is_some(), // <<< Only method name changed ...
        "cannot fetch classes; init level 'Servers' not yet loaded"
    );
```

**Comment by kang-sw** [comment]
@Bromeon

As the relevant PR #636 is merged, should I close this issue now, otherwise keep this open until upstream issue is fixed?

**Comment by Bromeon** [comment]
No, we can close this.
It's also possible to write in the initial PR message

> Closes #629

Then it's automated 🙂
